### ISSUE TITLE
Promotion with SArray * Number

### ISSUE THREAD

**Issue by KristofferC** [JuliaArrays/StaticArrays.jl#69]
This behavior is different between SVectors and base arrays (different resulting element type):

```jl
julia> 3.0 * rand(SVector{3, Float32})
3-element StaticArrays.SVector{3,Float64}:
 1.7314  
 0.394485
 0.566248

julia> 3.0 * rand(Float32, 3)
3-element Array{Float32,1}:
 1.5532  
 0.506801
 1.7563  
```

It is a bit annoying because it will promote the array when you multiply with a literal. In base they use `promote_array_type` it seems.

**Comment by SimonDanisch** [comment]
Good catch! This should most definitely be different! I think that's what has been called preserving the `container type`.

**Comment by andyferris** [comment]
Sigh. I wish this stuff was more obvious and intuitive (not that I disagree with the motivation, but just having a simple `broadcast` is quite valuable).

**Comment by KristofferC** [comment]
I think julia master just swapped to this behavior. Way to be ahead of the curve @andyferris :)

**Comment by SimonDanisch** [comment]
Awww, what? That's the most classic way for me to introduce type instabilities. Where was this decided?

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/pull/19692

**Comment by andyferris** [comment]
I'm going to close this. I don't intent to fix this for v0.5 if it matches v0.6 anyway.
